CONFIG.md

```diff
@@ -0,0 +1,419 @@
+2. Writing a SSLOptionsFactory.
+
+#### 1. Using builtin BasicSSLOptionsFactory
+Hazelcast Node.js Client includes a utility factory class that creates neccessary options object out of supplied
+properties. All you need to do is specify your factory as `BasicSSLOptionsFactory` and provide the following options.
+
+    certPath
+    caPath
+    servername
+    rejectUnauthorized
+    ciphers
+
+Refer to [`tls.connect` of Node.js](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback) for meaning
+of each option.
+> `certPath` and `caPath` defines file path to respective file that store such information.
+
+```json
+{
+    "network": {
+        "ssl": {
+            "enabled": true,
+            "factory": {
+                "exportedName": "BasicSSLOptionsFactory",
+                "properties": {
+                    "caPath": "ca.pem",
+                    "certPath": "cert.pem",
+                    "rejectUnauthorized": false
+                }
+            }
+        }
+    }
+}
+```
+If this options are not enough for your application, you may write you own options factory and instruct the client
+to get the options from it.
+
+#### Writing a SSL options factory
+In order to use full range of options provided to [`tls.connect` of Node.js](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback),
+you may write your own factory object.
+```json
+{
+    "network": {
+        "ssl": {
+            "enabled": true,
+            "factory": {
+                "path": "my_factory.js",
+                "exportedName": "SSLFactory",
+                "properties": {
+                    "caPath": "ca.pem",
+                    "certPath": "cert.pem",
+                    "keepOrder": true
+                }
+            }
+        }
+    }
+}
+```
+
+
+my_factory.js
+```javascript
+function SSLFactory() {
+}
+ 
+SSLFactory.prototype.init = function(props) {
+    this.keyPath = props.keyPath;
+    this.caPath = props.caPath;
+    this.keepOrder = props.userDefinedProperty1;
+};
+ 
+SSLFactory.prototype.getSSLOptions = function() {
+    var sslOpts = {
+        servername: 'foo.bar.com',
+        rejectUnauthorized: true,
+        cert: fs.readFileSync(this.keyPath),
+        ca: fs.readFileSync(this.caPath)
+    };
+    if (this.keepOrder) {
+        sslOpts.honorCipherOrder = true;
+    }
+    return sslOpts;
+}
+exports.SSLFactory = SSLFactory;
+```
+
+The client loads MyFactory.js at runtime and creates an instance of SSLFactory. It then, calls `init` method with
+the properties section in JSON config. Lastly, the client calls `getSSLOptions` method if SSLFactory to create the
+options object.
+
+For information about path resolution, refer to [Path Resolution](#path-resolution-and-object-loading)
+
+## Serialization configuration
+This section shows how to configure Hazelcast serialization declaratively. Please refer to [Hazelcast Docs](http://docs.hazelcast.org/docs/)
```

Comment:
...refer to [Hazelcast IMDG Reference Manual](http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#serialization)....